internal/provider/resources/block.go

```diff
@@ -130,6 +131,70 @@ func (r *BlockResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 	}
 }
 
+// getBlockSchemas fetches the block schemas for a given block type slug.
+//
+//nolint:ireturn // required by Terraform API
+func (r *BlockResource) getBlockSchemas(ctx context.Context, plan BlockResourceModel) ([]*api.BlockSchema, diag.Diagnostic) {
+	blockTypeClient, err := r.client.BlockTypes(plan.AccountID.ValueUUID(), plan.WorkspaceID.ValueUUID())
+	if err != nil {
+		return nil, helpers.CreateClientErrorDiagnostic("Block Types", err)
+	}
+
+	blockSchemaClient, err := r.client.BlockSchemas(plan.AccountID.ValueUUID(), plan.WorkspaceID.ValueUUID())
+	if err != nil {
+		return nil, helpers.CreateClientErrorDiagnostic("Block Schema", err)
+	}
+
+	blockType, err := blockTypeClient.GetBySlug(ctx, plan.TypeSlug.ValueString())
+	if err != nil {
+		return nil, helpers.ResourceClientErrorDiagnostic("Block Type", "get_by_slug", err)
+	}
+
+	blockSchemas, err := blockSchemaClient.List(ctx, []uuid.UUID{blockType.ID})
+	if err != nil {
+		return nil, helpers.ResourceClientErrorDiagnostic("Block Schema", "list", err)
+	}
+
+	return blockSchemas, nil
+}
+
+// getLatestBlockSchema fetches the latest block schema for a given block type slug.
+// If no block schemas are returned, the retrieval is retried because Prefect creates
+// them asynchronously after the creation of a workspace.
+//
+//nolint:ireturn // required by Terraform API
+func (r *BlockResource) getLatestBlockSchema(ctx context.Context, plan BlockResourceModel) (*api.BlockSchema, diag.Diagnostic) {
+	var blockSchemas []*api.BlockSchema
+	var latestBlockSchema *api.BlockSchema
+	var diags diag.Diagnostic
+
+	err := retry.Do(func() error {
```

Comment:
This is the special case in this PR where we need to retry block schemas because they're created asynchronously upon workspace creation.

For other resources, like workspace datasources, we can use`depends_on` to ensure a resource is created and ready prior to referencing it.